### PR TITLE
Cleanup-only PR, no functional changes

### PR DESCRIPTION
**Pull request by akshay-krishnan**
A bunch of things to clean up from the recent submitted PRs: 
- type hints were not consistent since we changed some things to delayed and back. 
- removed some duplicate code in two-view estimator
- removed some unused arguments
- ground truth poses get saved in the same colmap format with the estimated tracks (for potential visualization)

### CODE REVIEW

**Comment by travisdriver** [comment]
Would it be possible to hold off on merging this until I merge #619 and #603 (which I plan on doing today)? I'm afraid this will result in a lot of merge conflicts.

File: gtsfm/frontend/correspondence_generator/correspondence_generator_base.py
Comment (ayushbaid) [inline gtsfm/frontend/correspondence_generator/correspondence_generator_base.py:19]: Why make them delayed? I think we reverted to using non-delayed types during Hilti challenge.

File: gtsfm/runner/gtsfm_runner_base.py
Comment (ayushbaid) [inline gtsfm/runner/gtsfm_runner_base.py:219]: Do we need to pass in the full path of the config file?

**Review by ayushbaid — APPROVED** [review]

**Comment by ayushbaid** [comment]
Please enable mypy checks in CI if this PR resolves all of them.

**Comment by akshay-krishnan** [comment]
> Please enable mypy checks in CI if this PR resolves all of them.

There are too many to be resolved in a single PR :) These are only some outrageous errors that are fixed.

File: gtsfm/runner/gtsfm_runner_base.py
Comment (akshay-krishnan) [reply gtsfm/runner/gtsfm_runner_base.py]: Isnt that a better choice? Using output_root did not make sense at all, that is where outputs are stored. we could use the root folder + "configs/", but maybe its better to just ask the user to provide the full path.

File: gtsfm/frontend/correspondence_generator/correspondence_generator_base.py
Comment (akshay-krishnan) [reply gtsfm/frontend/correspondence_generator/correspondence_generator_base.py:19]: they are already delayed, I am only fixing the type hints here :)

Comment (akshay-krishnan) [reply gtsfm/frontend/correspondence_generator/correspondence_generator_base.py:19]: the reason we made them delayed in a different PR was so that they could be accessed on a cluster. I dont think they need to be delayed though.. that would be on @stepanyanhayk  to check.

File: gtsfm/runner/run_scene_optimizer_astrovision.py
Comment (travisdriver) [inline gtsfm/runner/run_scene_optimizer_astrovision.py:91]: Why did you remove this functionality?

Comment (travisdriver) [reply gtsfm/runner/run_scene_optimizer_astrovision.py:91]: Yes it is...

File: gtsfm/evaluation/metrics.py
Comment (johnwlambert) [inline gtsfm/evaluation/metrics.py:93]: just curious, where was this issue showing up?

File: gtsfm/evaluation/metrics.py
Comment (johnwlambert) [inline gtsfm/evaluation/metrics.py:319]: also curious, were some files not saving on some workers?

File: gtsfm/frontend/correspondence_generator/correspondence_generator_base.py
Comment (johnwlambert) [reply gtsfm/frontend/correspondence_generator/correspondence_generator_base.py:19]: Confirming that the image shapes to the CorrespondenceGenerator are currently delayed, according to this line: https://github.com/borglab/gtsfm/blob/master/gtsfm/runner/gtsfm_runner_base.py#L232

File: gtsfm/multi_view_optimizer.py
Comment (johnwlambert) [inline gtsfm/multi_view_optimizer.py:48]: Confirming that indeed the type is `Keypoints` and not `Delayed`, since the `CorrespondenceGenerator` landed.

https://github.com/borglab/gtsfm/blob/master/gtsfm/scene_optimizer.py#L247

File: gtsfm/multi_view_optimizer.py
Comment (johnwlambert) [inline gtsfm/multi_view_optimizer.py:55]: Confirming that the type is indeed Delayed, per

https://github.com/borglab/gtsfm/blob/master/gtsfm/scene_optimizer.py#L254

These come out of the TwoViewEstimator as Delayed objects:
https://github.com/borglab/gtsfm/blob/master/gtsfm/two_view_estimator.py#L382

File: gtsfm/evaluation/metrics.py
Comment (johnwlambert) [reply gtsfm/evaluation/metrics.py:93]: Conversion should already be automatic from None -> NaN if we are doing integer conversions for float:

```
>>> import numpy as np
>>> np.array([None, 1, 2, None])
array([None, 1, 2, None], dtype=object)
>>> x = np.array([None, 1, 2, None])
>>> y = np.array(x, dtype=np.float32)
>>> y
array([nan,  1.,  2., nan], dtype=float32)
```

but I guess we are using all ints here, and not casting to float32 with these int args?

File: gtsfm/view_graph_estimator/view_graph_estimator_base.py
Comment (johnwlambert) [inline gtsfm/view_graph_estimator/view_graph_estimator_base.py:166]: why note just do `plots_output_dir: Path = PLOT_BASE_PATH`, since line 191 effectively does this?

File: gtsfm/utils/io.py
Comment (johnwlambert) [inline gtsfm/utils/io.py:213]: @akshay-krishnan could you update the two docstring `Returns` items also?

File: gtsfm/two_view_estimator.py
Comment (johnwlambert) [inline gtsfm/two_view_estimator.py:263]: As far as I can tell, this last return arg should not be optional, and the docstring seems wrong in the ISP, since we never return a None report? https://github.com/borglab/gtsfm/blob/master/gtsfm/frontend/inlier_support_processor.py#L45

File: gtsfm/scene_optimizer.py
Comment (johnwlambert) [inline gtsfm/scene_optimizer.py:417]: nit: capitalization on "List of GT cameras"

File: gtsfm/scene_optimizer.py
Comment (johnwlambert) [inline gtsfm/scene_optimizer.py:409]: gt_gtsfm sounds a bit weird, how about `get_ground_truth_gtsfm_data()`?

File: gtsfm/scene_optimizer.py
Comment (johnwlambert) [inline gtsfm/scene_optimizer.py:475]: if possible, would be nice to have all 3 as keyword args, `gtsfm_data=gt_gtsfm_data, images=image_graph, ...`

https://github.com/borglab/gtsfm/blob/master/gtsfm/utils/io.py#L163-L165

File: gtsfm/scene_optimizer.py
Comment (johnwlambert) [reply gtsfm/scene_optimizer.py:409]: I wouldn't exactly call this GT, since the tracks can be noisy. I would call pure GT synthetic correspondences raycasted against mesh, w/ GT poses and GT intrinsics.

How about `get_gtsfm_data_w_est_tracks_and_gt_cameras()`?

File: gtsfm/scene_optimizer.py
Comment (johnwlambert) [inline gtsfm/scene_optimizer.py:410]: the GT cameras should be all or nothing for each of our datasets (either for every image, or none of them)

File: gtsfm/scene_optimizer.py
Comment (johnwlambert) [inline gtsfm/scene_optimizer.py:365]: from what I call, order of ops was very important for these alignment transforms. are we sure this still works out?

**Review by johnwlambert — APPROVED** [review]
LGTM

File: gtsfm/runner/run_scene_optimizer_astrovision.py
Comment (akshay-krishnan) [reply gtsfm/runner/run_scene_optimizer_astrovision.py]: Ive reverted this change, resolving

File: gtsfm/evaluation/metrics.py
Comment (akshay-krishnan) [reply gtsfm/evaluation/metrics.py:93]: I dont think we want to cast all metrics to float. there are some metrics that are assumed to be `int` for good reason. for example: the track length histogram in the report relies on all numbers being ints.

File: gtsfm/evaluation/metrics.py
Comment (akshay-krishnan) [reply gtsfm/evaluation/metrics.py:319]: I had problems with some metrics not being written to json because they could not be parsed. This was a way of not crashing the pipeline and knowing which metrics are causing the issue. I think its better than what we currently do, but also fine with not checking it in.

File: gtsfm/scene_optimizer.py
Comment (akshay-krishnan) [reply gtsfm/scene_optimizer.py:365]: yes, there are 2 alignments going on: 
1. aligning estimated poses to GT poses
2. aligning estimated poses so as to center the reconstruction

(1) was previously not happening for the saved gtsfm data. For the pose plots (which are not very usable for large datasets) (2) was happening before (1).

File: gtsfm/two_view_estimator.py
Comment (akshay-krishnan) [reply gtsfm/two_view_estimator.py:263]: thanks for catching that, it was annoying to have that optional around everywhere.

**Comment by akshay-krishnan** [comment]
@johnwlambert some 3-4 unresolved comments left, could you let me know if those need any changes?

**Pull request merged**
